### ISSUE TITLE
FileService:printStackTraceToFile:ERROR when validating a product with overlapping fields

### ISSUE THREAD

**Issue by fran-vallejo** [NASA-PDS/validate#379]
## 🐛 Describe the bug
Validating a product that contains a table with overlapping fields results in an error when validate tries to create a stack trace report file.

## 📜 To Reproduce
Run validate (either version 2.0.7 or 2.1.0-SNAPSHOT) using **mix_cal_hk_fpac_report_20181204** as target.

```
BepiVirtualBox:~/Software$ validate-2.0.7/bin/validate --target /home/fvallejo/Documents/mix_cal_hk_fpac_report_20181204.xml
FileService:printStackTraceToFile:ERROR: This function can only write to file that ends with .txt or .log

```

## 🕵️ Expected behavior
Validate runs and reports the product is not valid because it defines overlapping fields.

## 📚 Version of Software Used
validate 2.0.7 and validate 2.1.0-SNAPSHOT [repo commit 2cba1fd32441626e30669d6cd4fa1bc6f8088d5f]

## 🩺 Test Data / Additional context
Product [mix_cal_hk_fpac_report_20181204.zip](https://github.com/NASA-PDS/validate/files/6884980/mix_cal_hk_fpac_report_20181204.zip) attached. Fields 42 and 43 in this product overlap.

Please note that validate will report missing external references as well as the product contains references to elements not yet in PDS.

## 🖥 System Info
 - OS: Ubuntu 20.04
 - Java: OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2.20.04, mixed mode, sharing)

---

**Comment by jstone-psi** [comment]
This happens to me, also. I managed to track the problem to `TableDataContentValidationRule.java`, where the stack trace logger is trying to write to the data file instead of a log file based on the data file name.

@jordanpadams I think this can be fixed by changing

```
FileService.printStackTraceToFile(dataFile.getPath(),ex);
```

to

```          
FileService.printStackTraceToFile(dataFile.getPath() + ".log",ex);
```

in three places in `TableDataContentValidationRule.java`.

**Comment by qchaupds** [comment]
@jstone-psi @jordanpadams You are right, the call to printStackTraceToFile() should pass in either "null", or appends ".log".  The default name will be used if "null" is passed in.
With some changes to our Java code, we have success:

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 165 % validate -R pds4.label --skip-context-reference-check -r report_github379_label_invalid.json -s json -t src/test/resources/github379/mix_cal_hk_fpac_report_20181204.xml > & t2 

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 164 % egrep "message|type|severity"  report_github379_label_invalid.json
          "messages": [
              "severity": "ERROR",
              "type": "error.table.field_value_overlap",
              "message": "This field overlaps the next field. Current field ends at 348. Next field starts at 347"
    "messageTypes": [
        "messageType": "error.label.context_ref_not_found",
        "messageType": "error.table.field_value_overlap",

Code will be checkin at next opportunity.

**Comment by jordanpadams** [comment]
closed per #385
